llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -4217,3 +4220,25 @@ InstCombinerImpl::transformCallThroughTrampoline(CallBase &Call,
   Call.setCalledFunction(FTy, NestF);
   return &Call;
 }
+
+// op(select(%v, %x, %y), select(%v, %y, %x)) --> op(%x, %y)
+Instruction *
+InstCombinerImpl::foldCommutativeIntrinsicOverSelects(IntrinsicInst &II) {
+  if (!II.isCommutative()) {
```

Comment:
It has been checked by outer if statement. I think there should be an assertion here.

Reply:
Done